proxy/multi_app_conn_test.go

```diff
@@ -0,0 +1,80 @@
+package proxy
+
+import (
+	"errors"
+	"os"
+	"os/signal"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/mock"
+
+	abcimocks "github.com/tendermint/tendermint/abci/client/mocks"
+	"github.com/tendermint/tendermint/proxy/mocks"
+)
+
+func TestAppConns_Start_Stop(t *testing.T) {
+	quitCh := make(<-chan struct{})
+
+	clientCreatorMock := &mocks.ClientCreator{}
+
+	clientMock := &abcimocks.Client{}
+	clientMock.On("SetLogger", mock.Anything).Return()
+	clientMock.On("Start").Return(nil)
+	clientMock.On("Stop").Return(nil)
+	clientMock.On("Quit").Return(quitCh)
+
+	clientCreatorMock.On("NewABCIClient").Return(clientMock, nil).Times(4)
+
+	appConns := NewAppConns(clientCreatorMock)
+
+	appConns.Start()
+	clientMock.AssertCalled(t, "Start")
```

Comment:
We probably want to check that all connections have been started and stopped. One way to do this is to add `.Times(4)` onto the `On()` calls above, and then check `clientMock.AssertExpectations()` which asserts that all defined expectations have in fact been called the correct number of times (if given).